File: web-interface/unittest/xivojson/trunk.py
# -*- coding: utf-8 -*-
from __future__ import with_statement

__version__ = "$Revision$ $Date$"
__author__  = "Guillaume Bour <gbour@proformatique.com>"
__license__ = """
    Copyright (C) 2010-2011  Proformatique

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA..
"""
import cjson, os
from xivojson import *

"""REQUIREMENTS:
    - a context from-extern
    - a user with id 1 exists
"""

class Test07Trunk(XiVOTestCase):
    OBJ = 'sip'

    def setUp(self):
        super(Test07Trunk, self).setUp()
        self.client.register(self.OBJ, 'service/ipbx', 'trunk_management')


    def test_01_siptrunk(self):
        (resp, data) = self.client.list(self.OBJ)
        # no queue == code 204
        self.assertEqual(resp.status, 204)

        # ADD
        with open('xivojson/siptrunk.json') as f:
            content = cjson.decode(f.read())
        self.debug(content)	

        (resp, data) = self.client.add(self.OBJ, content)
        self.debug(data)
        self.assertEqual(resp.status, 200)

        # LIST / Check add
        (resp, data) = self.client.list(self.OBJ)
        self.assertEqual(resp.status, 200)

        data = cjson.decode(data)
        self.debug(data)

        self.assertEqual(len(data), 1)
        self.assertTrue('name' in data[0])
        self.assertTrue(data[0]['name'] == 'unittest')
        
        id = data[0]['id']

        # SEARCH
        (resp, data) = self.client.view(self.OBJ, id)
        self.assertEqual(resp.status, 200)

        data = cjson.decode(data)
        self.debug(data)
        self.assertTrue('trunkfeatures' in data)
        self.assertTrue(data['trunkfeatures']['protocol'] == 'sip')

        # DELETE
        (resp, data) = self.client.delete(self.OBJ, id)
        self.assertEqual(resp.status, 200)
        self.debug(data)

        # try to redelete => must return 404
        (resp, data) = self.client.delete(self.OBJ, id)
        self.assertEqual(resp.status, 404)


if __name__ == '__main__':
    unittest.main()

File: web-interface/src/tpl/www/bloc/service/ipbx/asterisk/dialaction/trunk.php
<?php

#
# XiVO Web-Interface
# Copyright (C) 2006-2011  Avencall
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

$form = &$this->get_module('form');
$url = &$this->get_module('url');
$dhtml = &$this->get_module('dhtml');

$element = $this->get_var('element');
$list = $this->get_var('destination_list','trunk');
$event = $this->get_var('event');

$linked = $this->get_var('dialaction',$event,'linked');
$action = $this->get_var('dialaction',$event,'action');

if(empty($list) === false):
	echo	'<div id="fd-dialaction-'.$event.'-trunk-actiontype" class="b-nodisplay">',
		$form->select(array('desc'	=> $this->bbf('fm_dialaction_trunk-actionarg1'),
				    'name'		=> 'dialaction['.$event.'][actionarg1]',
				    'labelid'	=> 'dialaction-'.$event.'-trunk-actionarg1',
				    'key'		=> 'identity',
				    'altkey'	=> 'id',
				    'invalid'	=> ($linked === false && $action === 'trunk'),
				    'default'	=> $element['dialaction']['actionarg1']['default'],
				    'selected'	=> $this->get_var('dialaction',$event,'trunk','actionarg1')),
			      $list),
		$form->text(array('desc'	=> $this->bbf('fm_dialaction_trunk-actionarg2'),
				  'name'	=> 'dialaction['.$event.'][actionarg2]',
				  'labelid'	=> 'dialaction-'.$event.'-trunk-actionarg2',
				  'size'	=> 10,
				  'value'	=> $this->get_var('dialaction',$event,'trunk','actionarg2')));
	echo	'</div>';
endif;

?>
